homeassistant/components/husqvarna_automower/number.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+from aioautomower.exceptions import ApiException
+from aioautomower.model import MowerAttributes
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AutomowerNumberEntityDescription(NumberEntityDescription):
+    """Describes Automower number entity."""
+
+    exists_fn: Callable[[MowerAttributes], bool] = lambda _: True
+    value_fn: Callable[[MowerAttributes], int]
+
+
+NUMBER_TYPES: tuple[AutomowerNumberEntityDescription, ...] = (
+    AutomowerNumberEntityDescription(
+        key="cutting_height",
+        translation_key="cutting_height",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.CONFIG,
+        native_min_value=1,
+        native_max_value=9,
+        value_fn=lambda data: data.cutting_height,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up number platform."""
+    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        AutomowerNumberEntity(mower_id, coordinator, description)
+        for mower_id in coordinator.data
+        for description in NUMBER_TYPES
+        if description.exists_fn(coordinator.data[mower_id])
+    )
+
+
+class AutomowerNumberEntity(AutomowerBaseEntity, NumberEntity):
+    """Defining the AutomowerNumberEntity with AutomowerNumberEntityDescription."""
+
+    entity_description: AutomowerNumberEntityDescription
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+        description: AutomowerNumberEntityDescription,
+    ) -> None:
+        """Set up AutomowerNumberEntity."""
+        super().__init__(mower_id, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{mower_id}_{description.key}"
+
+    @property
+    def native_value(self) -> float:
+        """Return the state of the number."""
+        return self.entity_description.value_fn(self.mower_attributes)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Change the value."""
+        try:
+            await self.coordinator.api.set_cutting_height(self.mower_id, value)
+        except ApiException as exception:
+            raise HomeAssistantError(
+                f"Command couldn't be sent to the command queue: {exception}"
+            ) from exception
```

Comment:
Let's make this a property in the entity description

Reply:
Sorry, I don't understand this. What do you mean exactly?

Reply:
Create a `set_value_fn` with the function to set the function with